.github/workflows/test.yml

```diff
@@ -0,0 +1,64 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        os: [ubuntu-18.04, ubuntu-16.04, macos-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Ubuntu cache
+        uses: actions/cache@v1
+        if: startsWith(matrix.os, 'ubuntu')
+        with:
+          path: ~/.cache/pip
+          key:
+            ${{ matrix.os }}-${{ matrix.python-version }}-${{
+            hashFiles('**/pyproject.toml') }}
+          restore-keys: |
+            ${{ matrix.os }}-${{ matrix.python-version }}-
+
+      - name: macOS cache
```

Comment:
It might be easier to set a global PIP cache env var `PIP_CACHE_DIR` to the same location on every platform

Reply:
Yes, that means less config. I'd kind of prefer to let pip do what it want at default locations but it's probably fine.

A GitHub dev on `actions/cache` said (although this applies more to `--cache-dir`):

> The only concern with setting the cache directory is that the user will have to determine where they want their cache, and requires setting the cache directory for every pip command they run (which could be easy to forget).

https://github.com/actions/cache/pull/86#issuecomment-553445259

What do you think? If `PIP_CACHE_DIR`, what location do you suggest?

(I'm hoping pip will add a `pip cache dir` to return its default directory and mean we don't need config in triplicate or setting `PIP_CACHE_DIR`: https://github.com/pypa/pip/issues/7350.

Reply:
ok, if you promise to come and make a PR to use `pip cache dir`, lets keep this how it is for now

Reply:
Will do, I've set a reminder!